python/lsst/daf/io/table/sql.py

```diff
@@ -0,0 +1,938 @@
+# -*- coding: utf-8 -*-
+"""
+Collection of query wrappers / abstractions to both facilitate data
+retrieval and to reduce dependency on DB-specific API.
+"""
+
+from __future__ import print_function, division
+from builtins import str, bytes, zip, range
+
+import numpy as np
+import sqlalchemy
+import struct
+
+from lsst.afw import table as afw_table
+from sqlalchemy import select
+from sqlalchemy import Table, Column, Numeric
+from sqlalchemy.schema import MetaData, CreateTable
+from sqlalchemy.types import (SmallInteger, Integer, BigInteger,
+                              Float, String, Boolean, Text)
+from sqlalchemy.types import to_instance, TypeEngine
```

Comment:
I've intentionally kept this one separate because one is specifically database types.